### PR TITLE
feat(test-kit): tests on new flow

### PR DESCRIPTION
**Pull request by barak007**

### CODE REVIEW

File: packages/core/src/com/initializers/webworker.ts [hunk 1]
```diff
@@ -5,17 +5,19 @@ import type { InitializerOptions } from './types.js';
...
-    workerOptions,
+    workerOptions = {},
+    workerExtension = '.js',
 }: WebWorkerInitializerOptions) {
-    const isModule = workerOptions?.type === 'module';
     const instanceId = communication.getEnvironmentInstanceId(env, endpointType);
-    const url = `${communication.getPublicPath()}${env}.webworker.${isModule ? 'mjs' : 'js'}${location.search}`;
+    const url = `${communication.getPublicPath()}${env}.webworker${workerExtension}${location.search}`;
     const webWorker = new Worker(url, {
+        type: (globalThis as any).DEFAULT_WORKER_TYPE || 'classic',
```
Comment (barak007) [inline packages/core/src/com/initializers/webworker.ts:20]: comment the story

File: packages/engine-cli/src/cli.ts [hunk 2]
```diff
@@ -1,24 +1,51 @@
...
+    const writeMetadataFiles = boolParam(args.get('writeMetadataFiles')) ?? !watch;
+
+    const feature = strParam(args.get('feature'));
+    const config = strParam(args.get('config'));
+    const publicPath = strParam(args.get('publicPath')) ?? '';
+    const engineConfigFilePath = strParam(args.get('engineConfigFilePath'));
+
+    if (help) {
+        console.log(engine.toString());
+        console.log('🤷‍♂️');
```
Comment (barak007) [inline packages/engine-cli/src/cli.ts:21]: return

File: packages/runtime-node/src/node-env-manager.ts [hunk 3]
```diff
@@ -164,9 +223,32 @@ export function createFeatureEnvironmentsMapping(
...
         }
-        if (envNames.length) {
-            featureToEnvironments[feature.scopedName] = envNames;
-        }
+        featureToEnvironments[feature.scopedName] = envNames;
     }
     return { featureToEnvironments, availableEnvironments };
 }
+
+async function _getPortFromManagerProcessInit(managerProcess: ChildProcess, featureName: string, configName: string) {
```
Comment (barak007) [inline packages/runtime-node/src/node-env-manager.ts:231]: remove

File: packages/scripts/src/create-node-entrypoint.ts [hunk 4]
```diff
@@ -70,12 +66,18 @@ export function createNodeEntrypoint({
...
 
 const options = workerData?.runtimeOptions ?? parseRuntimeOptions();
-console.log('[${env.name}]: Started with options: ', options);
+const verbose = options.get('verbose') ?? false;
+const envId = options.get('environment_id') ?? 'unknown_environment';
+if (verbose) {
+    console.log('[${env.name}]: Started with options: ', options);
+}
+
+bindMetricsListener();
```
Comment (barak007) [inline packages/scripts/src/create-node-entrypoint.ts:80]: unbind when there is error

File: packages/test-kit/src/engine-app-manager.ts [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+        }
+        return this.ready;
+    }
+    private async build() {
+        if (this.options.skipBuild) {
+            this.runMetadata = readMetadataFiles(OUTPUT_PATH);
+            return;
+        }
+        const engineConfig = await loadEngineConfig(process.cwd());
+        const build = this.options.skipBuild === false;
```
Comment (barak007) [inline packages/test-kit/src/engine-app-manager.ts:35]: build===true

**Review by plisovyi — APPROVED** [review]
👍🏼 🚀

**Pull request merged**